### PR TITLE
Pull/99

### PR DESCRIPTION
**Pull request by KostyaSha**
#99 with additional fixes

---
This change is [Reviewable](https://reviewable.io/reviews/kostyasha/yet-another-docker-plugin/101)

### CODE REVIEW

File: yet-another-docker-plugin/src/main/resources/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher/init.sh [hunk 1]
```diff
@@ -42,21 +42,40 @@ fi
...
 env # debug
 
-RUN_CMD="java -jar slave.jar"
-RUN_CMD+=" -noReconnect"
-RUN_CMD+=" -jnlpUrl ${JENKINS_URL}/${COMPUTER_URL}/slave-agent.jnlp"
-if [ ! -z "$COMPUTER_SECRET" ]; then
- RUN_CMD+=" -secret $COMPUTER_SECRET"
+RUN_CMD="java"
+if [ -n "$JAVA_OPTS" ] ; then
+   RUN_CMD+="$RUN_CMD $JAVA_OPTS"
```
Comment (gcbartlett) [inline yet-another-docker-plugin/src/main/resources/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher/init.sh]: Should be: `RUN_CMD="$RUN_CMD $JAVA_OPTS"`

File: yet-another-docker-plugin/src/main/resources/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher/init.sh [hunk 2]
```diff
@@ -65,17 +65,17 @@ env # debug
 
 RUN_CMD="java"
 if [ -n "$JAVA_OPTS" ] ; then
-   RUN_CMD+=" $JAVA_OPTS"
+   RUN_CMD+="$RUN_CMD $JAVA_OPTS"
```
Comment (gcbartlett) [inline yet-another-docker-plugin/src/main/resources/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher/init.sh]: Should be: `RUN_CMD="$RUN_CMD $JAVA_OPTS"`

**Comment by gcbartlett** [comment]
Note too, that unless your Docker image pre-creates the `jenkins` account (as mine did), an Alpine image will run into problems because it has `adduser` but not `useradd`.

**Comment by KostyaSha** [comment]
How i like this world :)  Then need to be adopted... Will fix later...

**Comment by clarsonneur** [comment]
Good catch, guys!

**Comment by clarsonneur** [comment]
@gcbartlett @KostyaSha I have a question:
If some uses their own image which is not alpine based, ie there is no `adduser`, How do you believe we should do?
detect where we are and call the appropriate one?

What do you think we could/should do?

**Comment by KostyaSha** [comment]
Yes, i think we need detect and use. In any case this JNLP launcher only one of possible way and anybody can implement their own launchers with other behaviours.  I'm busy but will try to end current enhancements and do release.

**Comment by clarsonneur** [comment]
Cool. Thank you.

**Comment by KostyaSha** [comment]
busybox wget doesn't support "no-check-certificate"

**Comment by clarsonneur** [comment]
Correct. The image must have wget installed, to make it work.

**Comment by KostyaSha** [comment]
Busybox has wget, but it has no options.

**Comment by clarsonneur** [comment]
Yes, and noway to disable checking certificate with default wget from busybox. So, on Alpine, the base image must have the wget package. I fixed that in my own image by installing it with :

`apk install wget -y`

So, I believe we should test it and report an error to install wget to fix it. May be the ignore certificate flag help should have the information, as well. Right?

**Comment by KostyaSha** [comment]
At least it may not fail under http connection that i have in test. We can also add piece of shell script into Launcher user configuration. That will allow to do manipulations before run without building image, though without logs it not very useful for debug.

**Comment by clarsonneur** [comment]
yeah

**Comment by clarsonneur** [comment]
But I would at least report a message in the init.sh that the wget need to be installed to make it work... So, the user know what to do.

**Comment by KostyaSha** [comment]
It has wget. I will try to look again at #84 if i found the way attach logs into, it will be easier debug.

**Comment by KostyaSha** [comment]
Test passed, merging.

**Comment by KostyaSha** [comment]
Thanks for contribution!

**Pull request merged**